src/lib.rs

```diff
@@ -122,15 +125,32 @@ mod platform {
             let task = task_for_pid(self.pid);
             if task.is_err() { return task.map(|_| ()) }
 
-            unsafe {
-                let mut read: mach_msg_type_number_t = 0;
+            let page_addr      = (addr as i64 & (-4096)) as mach_vm_address_t;
+	        let last_page_addr = ((addr as i64 + buf.len() as i64 + 4095) & (-4096)) as mach_vm_address_t;
+            let page_size      = last_page_addr as usize - page_addr as usize;
```

Comment:
could you add a comment explaining what's going on here? looks like the interface for `vm_read` is quite different from `process_vm_readv`. you have to read full pages?